From d06322e89ef2fa735088b6eeaca86daefe3f4be5 Mon Sep 17 00:00:00 2001
From: Yassin <yassin@Nixos>
Date: Mon, 24 Aug 2026 16:03:09 +0200
Subject: [PATCH 4/6] pinctrl: mediatek: mt6765: map virtual PMIC EINT

Add the virtual pin used to route EINT 144 from the pin controller to
the PMIC. MT6765 has 180 physical GPIOs; this descriptor is
interrupt-only.
---
 drivers/pinctrl/mediatek/pinctrl-mtk-mt6765.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/pinctrl/mediatek/pinctrl-mtk-mt6765.h b/drivers/pinctrl/mediatek/pinctrl-mtk-mt6765.h
index 772563720..dcbcf9fbd 100644
--- a/drivers/pinctrl/mediatek/pinctrl-mtk-mt6765.h
+++ b/drivers/pinctrl/mediatek/pinctrl-mtk-mt6765.h
@@ -1749,6 +1749,12 @@ static struct mtk_pin_desc mtk_pins_mt6765[] = {
 		DRV_GRP4,
 		MTK_FUNCTION(0, "GPIO179")
 	),
+	MTK_PIN(
+		180, "GPIO180",
+		MTK_EINT_FUNCTION(0, 144),
+		DRV_GRP4,
+		MTK_FUNCTION(0, NULL)
+	),
 };
 
 #endif /* __PINCTRL_MTK_MT6765_H */
